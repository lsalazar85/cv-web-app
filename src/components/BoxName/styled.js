import styled from 'styled-components';

const BoxNameContainer = styled.div`
    display: flex;  
    flex-direction: column;
    background: none;
    
    @media(min-width: 320px){
        width: 100%;
        padding: 15px 15px 15px 0; 
        justify-content: flex-start;
        background: none; 
    }

    @media(min-width: 992px) {
        width: 300px;
        padding: 0px;
        justify-content: space-between;
        background: #1c1c1c;
    }  

    @media(min-width: 1100px) { width: 480px; }
`

const BoxNameAvatar = styled.div`
    width: 100%;
    height: 100%;
    justify-content: center;
    align-items: center;

    @media(min-width: 320px){display:none;}
    @media(min-width: 992px){display:flex;}

`

const BoxNameContent = styled.div`
    display: flex; 
    color: #ffffff;
    line-height: 1.17;

    @media(min-width: 320px) {
        flex-direction: row;
        justify-content: space-between;
    }
    
    @media (min-width: 992px) {flex-direction: column; align-items: center;}

    h2 {
        font-size: calc(18px + (48 - 18) * ((100vw - 300px) / (1600 - 300)));
        margin-bottom: 0px; 
    }

    span {
        color: #00c483;
        margin: 24px 0;

        @media (min-width: 320px) {display: none; font-size: 18px;}
        @media (min-width: 992px) {display: block; font-size: 22px}
    }

    ul {
        display: flex;
        flex-direction: row;

        @media(min-width: 320px){margin-bottom: 0px;}
        @media(min-width: 992px){margin-bottom: 24px;}

        li {
            margin-right: 16px;

            &:last-child {
                margin-right: 0px;
            }

           a {
                color: #ffffff;
                opacity: 0.95;
                transition: 0.25s;
            
                &:hover{
                    color: #00c483;
                    opacity: 1;
                }

                svg {
                    @media (min-width: 320px) {width: 1.4em; height: 1.4em;}
                    @media (min-width: 992px) {width: 1.8em; height: 1.8em;}
                }
           }
        }
    }
`

export { BoxNameContainer, BoxNameContent, BoxNameAvatar }