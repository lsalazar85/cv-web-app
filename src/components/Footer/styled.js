import styled from 'styled-components';

const FooterContainer = styled.footer`
    
    @media (min-width: 320px){
        display: flex;
        flex-direction: row;
        display: none;
        position: absolute;
        bottom: 1.5%;

        span {
            display: inline-flex;
            flex-direction: row;
            align-items: center;
            font-size: 11px;

            svg {
                margin-left: 3px;

                &:first-child {
                    color: #64d1ee;
                }

                &:last-child {
                    color: #6e5493;
                    margin-top: 3px;
                }
            }
        }
    }
    
    @media (min-width: 992px){display: block;}
`

export { FooterContainer };