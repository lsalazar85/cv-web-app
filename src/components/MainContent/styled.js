import styled from 'styled-components';

const MainContentContainer = styled.div`
    display: flex;
    flex-direction: column;
    flex-shrink: 0;
    flex-grow: 1;
`

export { MainContentContainer }