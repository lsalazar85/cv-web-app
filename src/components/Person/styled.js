import styled from 'styled-components'

const PersonContainer = styled.div`
    width: 480px;
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    background-image: url('https://w5.foxdsgn.com/niko/black/wp-content/uploads/sites/4/2019/03/niko_img.jpg');
    background-repeat: no-repeat;
    background-position: center center;
    background-size: cover;
    opacity: 0.95;
    transition: 0.25s;

    @media(max-width: 1100px) {width: 300px;}

    @media(max-width: 992px) {
        background-image: none; 
        justify-content: flex-start;
        width: 100%;
        padding: 10px;
    }
`
const PersonName = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    color: #ffffff;
    line-height: 1.17;

    @media (max-width: 992px){
        display: flex; 
        flex-direction: row;
        justify-content: space-between;
    }

    h2 {
        font-size: calc(18px + (48 - 18) * ((100vw - 300px) / (1600 - 300)));
        @media(max-width: 992px) {margin-bottom: 0px;}
    }

    span {
        color: #00c483;
        margin: 24px 0;
        font-size: 22px;
        @media(max-width: 992px) {display: none;}
        @media(max-width: 1100px) {font-size: 18px;}
    }

    ul {
        display: flex;
        flex-direction: row;
        margin-bottom: 24px;
        @media(max-width: 992px) {margin-bottom: 0px;}

        li {
            margin-right: 16px;

            &:last-child {
                margin-right: 0px;
            }

           a {
            color: #ffffff;
            opacity: 0.95;
            transition: 0.25s;
            
            &:hover{
                color: #00c483;
                opacity: 1;
            }

            svg {
                width: 1.8em;
                height: 1.8em;
                @media(max-width: 992px) {
                    width: 1.4em;
                    height: 1.4em;
                }
            }
           }
        }
    }
`

export { PersonContainer, PersonName }