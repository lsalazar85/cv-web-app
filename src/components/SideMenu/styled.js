import styled from 'styled-components'

const SideMenuContainer = styled.aside`
    display: flex;
    flex-direction: column;
    padding: 0 32px;
    @media (max-width: 992px){
        border-right: 1px solid #3dba7d;
        padding: 0 15px;
    }
`
const SideIcons = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: ${props => props.position};
    align-items: center;
    height: 100vh;
    color: #ffffff;

    &:last-child{
        padding-bottom: ${props =>  props.position === 'flex-end' && '40px'}
    }

    svg {
        width: 2em;
        height: 2em;
        cursor: pointer;
        color: #ffffff;
        opacity: 0.95;
        transition: 0.25s;

        @media (max-width: 992px){
            width: 1.7em;
            height: 1.7em;
        }
        
        &:hover {
            color: #3dba7d;
            opacity: 1;
        }
    }

    a.active{
        svg {
            color: #3dba7d;
            transform: scale(1.20);
        }
    }

   ul {
        padding-top: 15em;
        @media (max-width: 992px){padding-top: 8em;}

        li {
            margin-bottom: 45px;

            &:last-child{
                margin-bottom: 0px;
            }
        }
   }
`


export { SideMenuContainer, SideIcons }