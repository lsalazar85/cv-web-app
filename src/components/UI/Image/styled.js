import styled from 'styled-components'

const ImageContainer = styled.div`
    width: ${props => props.width};
    height: auto;
    
    img {
        width: 100%;
        height: auto;
        border-radius:${props => props.circular && '50%'};
    }
`

export { ImageContainer }