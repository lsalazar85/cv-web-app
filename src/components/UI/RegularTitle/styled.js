import styled from 'styled-components'

const H2 = styled.h2`
    color: #ffffff;
    font-weight: 700;
    font-size: ${props => `calc(${props.minFontSize}px + (${props.maxFontSize} - ${props.minFontSize}) * ((100vw - 300px) / (1600 - 300)))`}
`

export { H2 };