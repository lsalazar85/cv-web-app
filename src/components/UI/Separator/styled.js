import styled from 'styled-components'

const Div = styled.div`
    width: 100%;
    height: 0.5px;
    background: #353535;
    margin-top: 24px;
`

export { Div }
