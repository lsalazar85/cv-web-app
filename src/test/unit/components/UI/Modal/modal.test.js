import React from 'react';
import { render, cleanup } from '@testing-library/react';

import Modal from '../../../../../components/UI/Modal';
import { Animation, ModalFooter } from '../../../../../components/UI/Modal/styled';

describe('Modal', () => {
  afterEach(cleanup);

  test('Modal should render correctly', () => {
    const { asFragment } = render(<Modal onClose={jest.fn()} show={true} />);

    expect(asFragment()).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  test('Modal should render correctly with others props', () => {
    const content = 'content';
    const footer = 'footer';

    const { asFragment } = render(
      <Modal
        onClose={jest.fn()}
        show={true}
        closeModal={true}
        title={'Modal Test File'}
        content={<div>{content}</div>}
        footer={<div>{footer}</div>}
      />,
    );

    expect(asFragment()).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});

// Styled Componentes Testing

describe('ModalFooter', () => {
  afterEach(cleanup);

  test('ModalFooter should render correctly', () => {
    const children = 'Modal';

    const { asFragment } = render(<ModalFooter position={'flex-end'}>{children}</ModalFooter>);

    expect(asFragment()).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Animation', () => {
  afterEach(cleanup);

  test('Animation should render correctly', () => {
    const content = 'Animation';
    const state = '';

    const { asFragment } = render(<Animation state={state}>{content}</Animation>);

    expect(asFragment()).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
