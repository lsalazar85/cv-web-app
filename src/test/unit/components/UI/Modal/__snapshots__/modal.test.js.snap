// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animation Animation should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx bRGPdV"
  >
    Animation
  </div>
</DocumentFragment>
`;

exports[`Modal Modal should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx cTdjxE"
  >
    <div
      class="sc-dlnjPT hgfUpP"
    />
    <div
      class="sc-hKFyIo ePWOzL"
      width="40rem"
    />
  </div>
</DocumentFragment>
`;

exports[`Modal Modal should render correctly with others props 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx cTdjxE"
  >
    <div
      class="sc-dlnjPT hgfUpP"
    />
    <div
      class="sc-hKFyIo ePWOzL"
      width="40rem"
    >
      <header
        class="sc-eCApGN hYhSt"
      >
        <h3>
          Modal Test File
        </h3>
      </header>
      <div
        class="sc-gtssRu gIbjpE"
        data-testid="close-modal"
      />
      <div
        class="sc-jSFkmK jdyFSc"
      >
        <div>
          content
        </div>
      </div>
      <footer
        class="sc-gKAblj bFFacn"
      >
        <div>
          footer
        </div>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ModalFooter ModalFooter should render correctly 1`] = `
<DocumentFragment>
  <footer
    class="sc-gKAblj fyPVfg"
  >
    Modal
  </footer>
</DocumentFragment>
`;
